server.js

```diff
@@ -2,34 +2,237 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import data from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+
+
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: {
+    type: String
+
+  },
+  authors: String,
+  average_rating: {
+    type: Number,
+    default: 0,
+  },
+  num_pages: {
+    type: Number,
+    default: 0
+  },
+  ratings_count: {
+    type: Number,
+    default: 0
+  },
+  text_reviews_count: {
+    type: Number,
+    default: 0
+  },
+  img_url: {
+    type: String,
+    default: ''
+  }
+})
+
+const User = mongoose.model('User', {
+  username: {
+    type: String,
+    minlength: 3,
+    maxlength: 20
+  },
+  password: {
+    type: String,
+    minlength: 8
+  },
+  ratings: {
+    type: Array,
+    default: []
+  },
+  profilePicture: {
+    type: String,
+    default: ''
+  }
+})
+
+
+const createRandomBook = async (num, titlesArray, authorArray) => {
+  await Book.deleteMany()
+  const arr = Array.apply(null, Array(num))
+
+  arr.forEach((val, index) => {
+    new Book({
+      bookID: index,
+      title: titlesArray[parseInt(titlesArray.length * Math.random())],
+      authors: authorArray[parseInt(authorArray.length * Math.random())],
+      num_pages: parseInt(Math.random() * 2000),
+      ratings_count: parseInt(Math.random() * 10000),
+      average_rating: (Math.random() * 5).toFixed(2)
+
+    }).save()
+  })
```

Comment:
Great job! 🥇 
There are a random user generator API's, Maybe you can make the public API from it too?